checkov/yaml_doc/base_registry.py

```diff
@@ -245,9 +254,43 @@ def get_result_key(self, check: BaseCheck,
                     if subconf_startline < start_line:
                         start_line = subconf_startline
             if start_line and end_line:
-                return f'{entity_type}.{entity_name}.{check.id}[{start_line}:{end_line}]'
-        return f'{entity_type}.{entity_name}.{check.id}'
+                key = f'{entity_type}.{entity_name}.{check.id}[{start_line}:{end_line}]'
+                return Registry.modify_gha_key(key, check, definition)
+
+        key = f'{entity_type}.{entity_name}.{check.id}'
+        return Registry.modify_gha_key(key, check, definition)
```

Comment:
instead of creating the `key` first and then do some splitting inside `modify_gha_key`, why not pass in the variable to actually create the key to a `create_gha_key()` method, which will return the correct key?